src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -280,8 +279,8 @@ protected internal virtual void WriteMetricEvents(SystemMetricEvent[] metricEven
             foreach (SystemMetricEvent metricEvent in metricEvents)
             {
                 _eventGenerator.LogFunctionMetricEvent(
-                    subscriptionId,
-                    appName,
+                    _environment.GetSubscriptionId() ?? string.Empty,
+                    GetNormalizedString(_environment.GetAzureWebsiteUniqueSlotName()),
```

Comment:
@brettsam Since metricslogger is no longer disposed during jobhost restarts, without this, metric logs seem to have the template site's subscription and appname even after specialization. Is there any other place I need to worry about having static appname and subsccriptionId?

Reply:
Chatted offline to use the IOptionsMonitor instead.